sql/parser/builtins.go

```diff
@@ -1250,6 +1252,28 @@ func round(x float64, n int64) (Datum, error) {
 	return DFloat(y), err
 }
 
+// returnTuple returns the Datum type that all arguments share, or an error
+// if they do not share types.
+func returnTuple(args DTuple) (Datum, error) {
+	var typ string
+	var i int
+	arg := DNull
+	// Find first non-null argument.
+	for ; i < len(args); i++ {
+		arg = args[i]
+		if arg != DNull {
+			typ = arg.Type()
+			break
+		}
+	}
+	for _, a := range args[i:] {
+		if a.Type() != typ && a != DNull {
```

Comment:
ok if they're all `NULL`?

Also, super nit: you're comparing `args[i]` with itself (but I don't think avoiding that is net positive).